src/main/java/org/wso2/carbon/connector/ISO8583/ISO8583MessageHandler.java

```diff
@@ -0,0 +1,142 @@
+            if (connection.isConnected()) {
+                outStream.writeUTF(isoMessage);
+                outStream.flush();
+
+                /* Sender will receive the Acknowledgement here */
+                if ((message = inFromServer.readLine()) != null) {
+                    unpackResponse(messageContext, message);
+                }
+            }
+        } catch (IOException e) {
+            handleException("An exception occurred in sending the ISO8583 message", e);
+        } finally {
+            try {
+                if (outStream != null) {
+                    outStream.close();
+                }
+                if (inFromServer != null) {
+                    inFromServer.close();
+                }
+                connection.close();
+            } catch (IOException e) {
+                log.error("Couldn't close the I/O Streams", e);
+            }
+        }
+    }
+
+    /**
+     * unpack the response string of isoMessage
+     *
+     * @param message  response String of isoMessage
+     * @param messageContext the message context
+     */
+    public void unpackResponse(MessageContext messageContext, String message) {
+        try {
+            ISOPackager packager = ISO8583PackagerFactory.getPackager();
+            ISOMsg isoMsg = new ISOMsg();
+            isoMsg.setPackager(packager);
+            isoMsg.unpack(message.getBytes());
+            messageBuilder(messageContext, isoMsg);
+        } catch (ISOException e) {
+            handleException("Couldn't unpack the message since message is not in ISO Standard :" + message, e);
+        }
+    }
+
+    /**
+     * Build the response message in xml format and set to message context
+     *
+     * @param isomsg  response fields  of isoMessage
+     * @param messageContext the message context
+     */
+    public void messageBuilder(MessageContext messageContext, ISOMsg isomsg) {
+        OMFactory OMfactory = OMAbstractFactory.getOMFactory();
+        OMElement parentElement = OMfactory.createOMElement(ISO8583Constant.TAG_MSG, null);
+        OMElement result = OMfactory.createOMElement(ISO8583Constant.TAG_DATA, null);
+        for (int i = 0; i <= isomsg.getMaxField(); i++) {
+            if (isomsg.hasField(i)) {
+                String outputResult = isomsg.getString(i);
+                OMElement messageElement = OMfactory.createOMElement(ISO8583Constant.TAG_FIELD, null);
+                messageElement.addAttribute(OMfactory.createOMAttribute(ISO8583Constant.TAG_ID, null, String.valueOf(i)));
+                messageElement.setText(outputResult);
+                result.addChild(messageElement);
+                parentElement.addChild(result);
+                messageContext.getEnvelope().getBody().addChild(parentElement);
+            }
+        }
+    }
+
+    /**
+     * handle the Exception
+     *
+     * @param  msg error message
+     * @param ex an Exception
+     */
+    private void handleException(String msg, Exception ex) {
```

Comment:
In this ISO8583MessageHandler class , I didn't extends any  classes and  ISO8583MessageHandler is like an util class.